src/realm/object-store/class.hpp

```diff
@@ -0,0 +1,56 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_CLASS_HPP
+#define REALM_OS_CLASS_HPP
+
+#include <realm/obj.hpp>
+#include "realm/parser/keypath_mapping.hpp"
+#include "realm/parser/query_parser.hpp"
+#include <memory>
+
+namespace realm {
+
+class ObjectSchema;
+class Realm;
+class Query;
+
+class Class {
+public:
+    Class(const std::shared_ptr<Realm>& r, StringData object_type);
+
+    size_t size() const;
+    TableKey get_key() const;
+    ColKey get_column_key(StringData) const;
+    Query get_query(const std::string& query_string, query_parser::Arguments& args,
+                    const query_parser::KeyPathMapping& mapping) const;
+
+    Obj create_object(Mixed primary_value = {});
```

Comment:
I noticed that we only use `create_object()` for objects without primary keys. When there are primary keys, we use this helper instead:

```cpp
    static std::pair<Obj, bool> get_or_create_object_with_primary_key(TableRef table, const Mixed& primary_key)
    {
        bool did_create;
        auto obj = table->create_object_with_primary_key(primary_key, &did_create);
        return {obj, did_create};
    };
```

It would be great if this was the API exposed.

Reply:
:+1: